.github/actions/slack-notification/slack_payload.py

```diff
@@ -0,0 +1,155 @@
+"""
+Slack Notification Script
+
+This script processes environment variables to generate a Slack-compatible message.
+It supports two modes:
+1. If "MESSAGE" is set, it processes the message using the 'process_message_env' function.
+2. If "PAYLOAD" is set, it directly loads the payload as JSON.
+
+Environment Variables:
+- "MESSAGE": The text content for the Slack message.
+- "PAYLOAD": A JSON-formatted payload for the Slack message.
+- "URL_ENCODED": If set to 'true', it URL-decodes the "MESSAGE".
+- "GITHUB_OUTPUT": The file path for GitHub Actions workflow output.
+- "MASK": If set to 'true', masks sensitive information in the output.
+
+The resulting JSON data is either written to the specified output file for GitHub Actions workflow
+or printed with masking if "MASK" is set to 'true'.
+"""
+
+import json
+import os
+from urllib.parse import unquote
+
+
+class SlackException(Exception):
+    """
+    Custom exception class for errors related to Slack notifications.
+    """
+
+
+def create_message_block(message):
+    """
+    Creates a message block for Slack with the given text.
+
+    Parameters:
+    - message (str): The text content for the message block.
+
+    Returns:
+    dict: A dictionary representing the Slack message block in the following format:
+          {
+              "type": "section",
+              "text": {
+                  "type": "mrkdwn",
+                  "text": message
+              }
+          }
+    """
+    return {
+        "type": "section",
+        "text": {
+            "type": "mrkdwn",
+            "text": message,
+        },
+    }
+
+
+def process_message_env():
+    """
+    Process the 'MESSAGE' environment variable for creating a Slack-compatible message.
+
+    If 'URL_ENCODED' is set to 'true', the message is URL-decoded.
+    The message is then formatted as a Slack message containing a single text block.
+
+    Returns:
+    dict: A dictionary representing the Slack message with the processed 'MESSAGE'.
+          The dictionary has the following format:
+          {
+              "text": processed_message,
+              "blocks": [
+                  {
+                      "type": "section",
+                      "text": {
+                          "type": "mrkdwn",
+                          "text": processed_message
+                      }
+                  }
+              ]
+          }
+    """
+    message = os.environ.get("MESSAGE", "No message")
```

Comment:
if there's no message you'll keep doing all the logic on the `"No message"` string

Reply:
Added an `if` statement to skip logic for `'no message'`.

Reply:
removed "No message"

Reply:
```
message = os.environ.get("MESSAGE", "No message")
if message != "No message":
    ...
```

I suggest moving this code and the following lines in a new function and adding the informative comment you typed in the other thread as a docstring:
> Sure. This section is relevant to sending a simple message. In the action, we are utilizing `GITHUB_OUTPUT` to output the message for Slack notification. The challenge arises when the message is a multiline string. When attempting to write to `GITHUB_OUTPUT`, GitHub Actions notifies about formatting problems with the string. There are several solutions to work around this issue. One approach is to use delimiters between lines, as suggested by [GitHub Workflows](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings). However, the GitHub solution warns that the delimiter chosen should not occur on a line of its own within the value, and this can be problematic for arbitrary multiline messages. To avoid this, another solution involves encoding the string. Therefore, if a multiline encoded string is provided, it should be decoded first. In this case, the encoding of the string was selected to handle the multiline format, and the last line of code normalizes the string by removing leading and trailing whitespaces from each line.

Reply:
The code already has a dedicated function for handling messages. The documentation has been updated.